### PR TITLE
Support simple case expression evaluation.

### PR DESCRIPTION
**Pull request by petermattis**
Rewrite the range cond evaluation to only evaluate the "left" expression
once.

### CODE REVIEW

**Comment by tamird** [comment]
left some comments on the commits, so don't rebase yet!

File: sql/parser2/eval.go [hunk 1]
```diff
@@ -612,6 +632,10 @@ func evalComparisonOp(op ComparisonOp, left, right Datum) (Datum, error) {
 		// GE(left, right) is implemented as LE(right, left)
 		op = LE
 		left, right = right, left
+	case NotIn:
```
Comment (spencerkimball) [inline sql/parser2/eval.go]: Where's the case for `In`?

Comment (spencerkimball) [inline sql/parser2/eval.go]: Ah scratch that. I see how it works.

File: sql/parser2/eval.go [hunk 2]
```diff
@@ -668,20 +687,41 @@ func evalFuncExpr(expr *FuncExpr, env Env) (Datum, error) {
...
 func evalCaseExpr(expr *CaseExpr, env Env) (Datum, error) {
 	if expr.Expr != nil {
-		// TODO(pmattis): These are expressions of the form `CASE <val> WHEN <val>
-		// THEN ...`.
-		return null, fmt.Errorf("TODO(pmattis): unsupported simple case expression: %T", expr)
-	}
-
-	for _, when := range expr.Whens {
-		d, err := EvalExpr(when.Cond, env)
+		// CASE <val> WHEN <expr> THEN ...
```
Comment (tschottdorf) [inline sql/parser2/eval.go]: is there `ELSE` as in `CASE [<val>] WHEN <[bool-]expr> THEN <expr> ELSE <expr>`?

Comment (tschottdorf) [inline sql/parser2/eval.go]: currently anything that would hit `ELSE` gets a `bool(true)` back, correct?

File: sql/parser2/eval.go [hunk 3]
```diff
@@ -710,3 +750,43 @@ func evalTupleEQ(ldatum, rdatum Datum) (Datum, error) {
...
 }
+
+func evalTupleIN(arg, values Datum) (Datum, error) {
+	if arg == null {
+		return dbool(false), nil
+	}
+
+	vtuple := values.(dtuple)
+
+	if _, ok := arg.(dtuple); !ok && false {
```
Comment (tschottdorf) [inline sql/parser2/eval.go]: comment here about the dead code path.

File: sql/parser2/eval.go (hunk 2 above)
Comment (tschottdorf) [inline sql/parser2/eval.go]: nevermind, I was missing a few lines of the function in the diff.

**Comment by tschottdorf** [comment]
LGTM

**Comment by tamird** [comment]
LGTM modulo comment on `&& false`

File: sql/parser2/eval.go (hunk 3 above)
Comment (petermattis) [inline sql/parser2/eval.go]: I thought that's what the comment on the next line was about. Guess I need to make it clearer.

**Pull request merged**